tests/playwright/src/model/pages/resources-page.ts

```diff
@@ -23,16 +23,10 @@ import { SettingsPage } from './settings-page';
 export class ResourcesPage extends SettingsPage {
   readonly heading: Locator;
   readonly featuredProviderResources: Locator;
-  readonly podmanResources: Locator;
-  readonly composeResources: Locator;
-  readonly kindResources: Locator;
 
   constructor(page: Page) {
     super(page, 'Resources');
     this.heading = this.header.getByRole('heading', { name: 'Title' }).and(this.header.getByText('Resources'));
     this.featuredProviderResources = this.content.getByRole('region', { name: 'Featured Provider Resources' });
-    this.podmanResources = this.featuredProviderResources.getByRole('region', { name: 'podman', exact: true });
```

Comment:
You can add functions for checking if provider connection/cli is actually visible/available and also a function that return particular page object. Similarly as in the `ExtensionsPage`.

Reply:
I attempted to implement this function, but I find it more complicated than initializing an object in a specific scenario. We need a generic function that can return two types of objects (cli or content).  The function definition should look like this:

```
async getResourceCardPage<T>(
  resourceName: string,
  type: new (page: any, resourceName: string, resourceElementVisibleName?: string) => T,
  resourceElementVisibleName?: string
): Promise<T> 
```

Reply:
Why not implement an interface and return the interface?

Reply:
Unfortunately, I am not sure how to implement such a solution. @cbr7, could you please provide a more detailed description?

Reply:
What I mean is that the method does not need to be generic at all, all it needs is to return an interface type which is implemented in all the POM classes that the the method could refer too, for example:

`async getResourceCardPage(resourceName: string, resourceElementVisibleName?: string): Promise<nameOfInterface>`

And all the method can return different POM classes, potentially from a switch statement, as long as all the POM classes that it can return implement nameOFInterface

Reply:
I don't think using an interface is a good idea due to the different properties in POM classes. For example, in:

```typescript
export class ResourceConnectionCardPage extends ResourceCardPage implements IResourceCardPage {
  readonly resourceElement: Locator;
  readonly resourceElementDetailsButton: Locator;
  readonly resourceElementConnectionStatus: Locator;
  readonly resourceElementConnectionActions: Locator;
  ...
}

```

I have specific properties that need to be added to the interface. However, if I include them as optional properties like `resourceElementConnectionActions?: Locator`, it can cause issues when I need to use them in specs because they may be undefined. If I don't make them optional, I would have to implement them in all POM classes, even when they aren't needed. 

I would like to keep my current solution, but maybe I just didn't understand your intentions. Please let me know what you think.

Reply:
I'm not sure what the issue is, if you need to use them somewhere you need to check that they are defined, this is true for all properties in all classes so I'm not sure how it's any different here.

Don't take this the wrong way, but generic methods are extremely difficult to debug and should not be used unless absolutely required and they aren't required here.

Additionally if you dont want to have fields defined in the interface, that is fine, don't have them defined there, have them only in the POM, nobody says the interface itself must have them, all you need is to cast the resulting object that is returned from the interface, which will be of Type interface into whatever POM class you know it is.

Reply:
I think we have a little misunderstanding. :) I don’t use a generic function; I use https://github.com/containers/podman-desktop/blob/49508fb2696dfff445ccc3680d73c95f50272363/tests/playwright/src/specs/compose-onboarding-smoke.spec.ts#L77. If you think that using a function with an interface would be better, I can try that approach again.

Reply:
I was talking about you post where you pasted:

```
async getResourceCardPage<T>(
  resourceName: string,
  type: new (page: any, resourceName: string, resourceElementVisibleName?: string) => T,
  resourceElementVisibleName?: string
): Promise<T> 
```

Which is why I used the name method name when I gave you an example of using the interface for a return type.